src/librawspeed/decoders/CrwDecoder.cpp

```diff
@@ -193,8 +193,9 @@ void CrwDecoder::decodeMetaDataInternal(CameraMetaData *meta) {
       ushort16 wb_index = shot_info->getU16(7);
       CiffEntry *wb_data = mRootIFD->getEntryRecursive(CIFF_WHITEBALANCE);
       /* CANON EOS D60, CANON EOS 10D, CANON EOS 300D */
-      int wb_offset = (wb_index < 18) ? "0134567028"[wb_index]-'0' : 0;
-      wb_offset = 1+wb_offset*4;
+      if (wb_index > 9)
+        ThrowIOE("CrwDecoder: invalid white balance index");
```

Comment:
The type of the exception might be questionable (as are _all_ exception types). The fact is, if `wb_index > 9` then the file is broken.